### PR TITLE
Update to use latest HttpClientHandler which has improved certificate support

### LINKED ISSUE #960: Unable to specify a client certificate when choosing manual option.
HttpClientHandler needs to support the ClientCertificate property in order to set a client certificate when choosing the manual ClientCertificateOptions.
Issue opened in corefx repo: dotnet/corefx#4479

Failing test case...
FunctionalTest.Security.TransportSecurity.Tests#HttpsTests.ClientCertificate_EchoString

### PR DESCRIPTION
**Pull request by mconnew**
This addresses issue #959 for all platforms except UWP
This is a step needed to make these tests work on CI machines
Fixes #960

### CODE REVIEW

**Comment by mconnew** [comment]
I may need to add active issue #1123 back in on the tests as I'm unsure if we run the certificate setup on our CI machines. This PR will verify this.
Manually verified this fix on Ubuntu

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

File: src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh [hunk 1]
```diff
@@ -122,26 +135,26 @@ else
...
-    echo "Certificate will be obtained from BridgeHost '$__bridge_host'"
-    if [ -n "$BridgeHost" ]; then 
-        echo "WARNING: This will replace the BridgeHost specified in the environment, '$BridgeHost', for the duration of this script"
+if [ -n "$__service_host" ]; then 
+    echo "Certificate will be obtained from ServiceHost '$__service_host'"
+    if [ -n "$ServiceUri" ]; then 
+        echo "WARNING: This will replace the ServiceUri specified in the environment, '$ServiceUri', for the duration of this script"
     fi 
     
     # set BridgeHost as an external env var so that this can be picked up by the Bridge
```
Comment (roncain) [inline src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh]: Nit: comment is now obsolete and should mention ServiceUri

File: src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh [hunk 2]
```diff
@@ -3,13 +3,13 @@
 show_banner()
```
Comment (roncain) [inline src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh]: Nit: consider renaming BridgeCertificateInstaller.sh and BridgeCertificateInstaller.cs.  Not required.

**Comment by roncain** [comment]
@dotnet-bot test outerloop Ubuntu14.04

**Comment by roncain** [comment]
LGTM, with only some non-blocking questions:
- @mconnew -- in testing this manually, were you following our manual testing rd doc steps, and if so were they accurate (especially related to the running of this script)?
- @iamjasonp and @mconnew -- could one of you help me understand why these tests passed in CI on *nix?  If you look at console output (search for "root cert") you'll see the tests successfully acquired the cert during test running (which, incidentally will expire in 1 week) and installed it into the store.  Because CI did not run the BridgeCertificateInstaller.sh, I now don't understand why that script is needed at all. @iamjasonp could you help explain why we passed CI without running this script and whether we still need it?

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.CoreClr.cs [hunk 3]
```diff
@@ -10,10 +10,10 @@
 
 namespace System.ServiceModel.Channels
 {
-#if FEATURE_CORECLR && !TARGETS_WINDOWS
+#if FEATURE_CORECLR
     public partial class ServiceModelHttpMessageHandler
     {
-        private HttpClientHandler _innerHandler;
+        private readonly HttpClientHandler _innerHandler;
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.CoreClr.cs]: For future, consider making the same "readonly" change to the NetNative HttpMessageHandler.

File: src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/project.json [hunk 4]
```diff
@@ -6,7 +6,9 @@
     "System.IO.FileSystem": "4.0.0",
     "System.IO": "4.0.10",
     "System.Net.Http": "4.1.0-rc3-24109-00",
-    "System.Net.Primitives": "4.0.10"
+    "System.Net.Primitives": "4.0.10",
+    "System.ServiceModel.Http": "4.1.0-rc3-24109-00",
+    "System.ServiceModel.Primitives": "4.1.0-rc3-24109-00"
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/project.json]: Will this work? I feel that we'll probably need to rev the package to a later package anyway to take this change. We can merge this now though.

File: src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh (hunk 1 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh]: I want to change the name of ServiceUri as it doesn't contain a Uri, just a hostname. That will need to wait for another time. I'll change the comment.

**Comment by mconnew** [comment]
@dotnet-bot test outerloop Ubuntu14.01

File: src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh [hunk 5]
```diff
@@ -122,26 +135,26 @@ else
...
+if [ -z "$ServiceUri" ]; then
+    echo "WARNING: \$ServiceUri environment variable was not specified in the environment"
     echo "         This may be caused by running under sudo; run 'sudo -E' if you want to keep the user environment"
     echo ""
-    echo "         Either set the BridgeHost environment variable, or run '$0' with a --bridge-host parameter"
+    echo "         Either set the ServiceUri environment variable, or run '$0' with a --service-host parameter"
     echo ""
-    echo "         The default BridgeHost is 'localhost'."
+    echo "         The default ServiceUri is 'localhost'."
     echo "         If the Bridge is running on a port other than the default, also set the 'BridgePort' variable"
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tools/setupfiles/BridgeCertificateInstaller.sh]: NIT: Please also change this comment referring to the Bridge

File: src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/CertificateInstaller.cs [hunk 6]
```diff
@@ -64,66 +66,64 @@ public static int Main(string[] args)
...
-
-                Console.WriteLine(certAsPemUri);
-                Console.WriteLine();
-
-                HttpClient client = new HttpClient();
-                var response = client.GetAsync(certAsPemUri).GetAwaiter().GetResult();
+            factory = new ChannelFactory<IUtil>(basicHttpBinding, new EndpointAddress(address));
+            serviceProxy = factory.CreateChannel();
+            certificateBytes = serviceProxy.GetRootCert(true);
+            Console.WriteLine("    ... read {0} bytes from Bridge", certificateBytes.Length);
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/CertificateInstaller.cs]: NIT: "bridge"

Comment (mconnew) [inline src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/CertificateInstaller.cs]: I'm going to change it to "test service" but we need to come up with a clean consistent name.

**Comment by iamjasonp** [comment]
LGTM.

File: src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/project.json (hunk 4 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/project.json]: I don't follow what you are saying.

Comment (iamjasonp) [inline src/System.Private.ServiceModel/tools/test/BridgeCertificateInstaller/project.json]: Ah I missed that you're using a WCF proxy here. 

Does BridgeCertificateInstaller require the latest S.SM.Primitives? If not, no need to bump it up to a version that's this high

**Comment by mconnew** [comment]
I needed to rebase because of package version changes so rerunning the outerloop build for windows.
@dotnet-bot test outerloop Windows_NT

**Pull request merged**